synapse/visibility.py

```diff
@@ -119,22 +118,26 @@ def allowed(event):
 
                the original event if they can see it as normal.
         """
-        if event.type == "org.matrix.dummy_event":
+        if event.type == "org.matrix.dummy_event" and filter_send_to_client:
```

Comment:
Would it make sense to have this whole block in a `if filter_send_to_client:` block which then has the 4 if-statements in it?

Reply:
At this point, probably, yes

Reply:
Done in https://github.com/matrix-org/synapse/pull/7066/commits/8120a238a465de576ad4d171e3072b28e5df32ac